server/dates.ts

```diff
@@ -0,0 +1,38 @@
+import { NextApiHandler, NextApiRequest, NextApiResponse } from 'next';
+import { connectToDatabase } from './mongoDB';
+import { SingletonDefinition } from '../common/types';
+import { isAdmin } from './protect';
+
+export const getDate = async (
+  req: NextApiRequest,
+  res: NextApiResponse,
+  dateName: string
```

Comment:
instead of having this as type `string`, (w the above changes) it could be type `DateSingleton["type"]` or `SingletonType.RegistrationOpen | SingletonType.RegistrationClosed | SingletonType.ConfirmBy` (or you could also define an alias for the latter in types.ts and refer to it both here and in the definition for SingletonType).